spec.html

```diff
@@ -7504,9 +7504,10 @@ <h1>MakeMethod ( _F_, _homeObject_ )</h1>
       <h1>SetFunctionName ( _F_, _name_ [ , _prefix_ ] )</h1>
       <p>The abstract operation SetFunctionName requires a Function argument _F_, a String or Symbol argument _name_ and optionally a String argument _prefix_. This operation adds a `name` property to _F_ by performing the following steps:</p>
       <emu-alg>
-        1. Assert: _F_ is an extensible object that does not have a `name` own property.
+        1. Assert: _F_ is an extensible object.
         1. Assert: Type(_name_) is either Symbol or String.
         1. Assert: If _prefix_ is present, then Type(_prefix_) is String.
+        1. If ! HasOwnProperty(_F_, `"name"`) is *true*, return.
```

Comment:
I’m not sure it’s as clear to have an operation named “set function name” not always set the function name, although i see how this DRYs up the steps. What do you think?

Reply:
The status quo has '?' before the corresponding calls to `HasOwnProperty`. I don't think you're justified in changing that to '!' here.

Reply:
I'm wondering the same thing as @ljharb. I like the general idea of cleaning things up with refactoring, but for specs we should prioritize clarity.